File: medium/8. First Duplicate Value/first_duplicate_value.go
package main

import "math"

func FirstDuplicateValue(array []int) int {
	for i := 0; i < len(array); i++ {
		if array[int(math.Abs(float64(array[i])))-1] < 0 {
			return int(math.Abs(float64(array[i])))
		}
		array[int(math.Abs(float64(array[i])))-1] *= -1
	}
	return -1
}

File: medium/8. First Duplicate Value/first_duplicate_value.cpp
#include <unordered_set>
#include <vector>
#include <math.h>

int firstDuplicateValue(std::vector<int> array) {
  for (int i = 0; i < array.size(); i++) {
    if (array[std::abs(array[i]) - 1] < 0) {
      return std::abs(array[i]);
    }
    array[std::abs(array[i]) - 1] *= -1;
  }
  return -1;
}
